### ISSUE TITLE
Content Security Policy (CSP) Issue with React API Reference component

### ISSUE THREAD

**Issue by michaelg-baringa** [scalar/scalar#3973]
### What happens?

I have created a minimal example to demonstrate the CSP issue (https://github.com/michaelg-baringa/scalar-example/blob/main/nextjs-blog/README.md)

When navigating to http://localhost:3001/docs, only some of the scalar components load and the browser console shows many CSP errors

### What did you expect to happen?

The scalar docs page to load

### OpenAPI Document

https://cdn.jsdelivr.net/npm/@scalar/galaxy/dist/latest.json

**Comment by michaelg-baringa** [comment]
I've been told that I just need to add the sha256 hashes of the relevant elements to the middleware.ts. Using the Chrome browser, it will calculate the hashes for you and display that in the console error, whereas with Firefox it wasn't so I was struggling there.   

So I would say this isn't a bug and more of an ask for advice. Is this best practice, to add the sha's to the CSP header or is it possible to get the nonce added? I just feel that if something changes on scalar's side, then that might change one of the sha's and then it would break

**Comment by bensmith2093** [comment]
I have also seen this issue - unfortunately depending on what client calls are made within the application, different hashes are presented. They also do not seem to be particularly finite in nature!

Is there an option to allow nonce injection, or prevent additional hashes from being created depending on what actions the user performs?

**Comment by melroyvandenberg** [comment]
Same issue with Firefox.

**Comment by kishore03109** [comment]
This project looks promising, but not being able to add nonce or some SHA to the CSP header so that CSP does not get bypassed is a blocker for using this for production applications. Would appreciate this is prioritised!

**Comment by leixuexxxx** [comment]
Any update for this? Adding a `unsafe-inline` CSP header is not an ideal solution.

**Comment by Monyster** [comment]
Any updates? I have the same problem within Google Chrome. I am using the csp unblock extension. How can I unblock it without creating new security issues?

**Comment by SuperManifolds** [comment]
It is absurd that this is not being addressed. Why are you using unsafe eval in a product you are selling to enterprise customers?

**Comment by kieraneglin** [comment]
+1 - this is stopping us from using Scalar, unfortunately

Edit: although we can't use it, here's how to set it up to work with Rails:

```ruby
class DocsController < ApplicationController
  before_action :allow_inline_styles_without_nonce, only: :index

  content_security_policy do |policy|
    policy.style_src(*policy.style_src, :unsafe_inline)
  end

  def index
    # ...
  end

  private

  def allow_inline_styles_without_nonce
    request.content_security_policy_nonce_directives = ["script-src"]
  end
end
```

**Comment by lockinmarv** [comment]
Yeah pretty weird they doesn't address this issue. I fixed it by downgrading to v1.26.2

**Comment by marcelscr** [comment]
+1 - also waiting for a fix!

**Comment by Slicksim** [comment]
I have used a rewriter to add the nonce to the script tags, but you still need unsafe-inline on style-src-elem and style-src-attr.  And despite the nonce being added to the script tags, there is still an unsafe-eval failure.

On a pen test, these will be flagged as being insecure which is a bad look for any production software

+1 for adding support for getting this to work under a strict and secure CSP

**Comment by FranciscoKnebel** [comment]
ran into this issue today. I don't want to enable unsafe-inline, so I added the sha hash directly, and that seems to have fixed it.

**Comment by wenzf** [comment]
downdgrading to `@scalar/api-reference-react": "^0.7.7` fixed the issue for me

**Comment by amritk** [comment]
Alright I will be updating the react packages soon, so I'll take a look!

**Comment by codingespresso** [comment]
@hanspagel 
Unfortunately we have also encountered this issue while evaluating scalar and it is a showstopper for us.
Could you provide us with a timeline when this is going to be addressed?

**Comment by DemonHa** [comment]
is this fixed @amritk on the new react package ?
